### PR TITLE
ENH: Added type annotations to eye() function

### LINKED ISSUE #18472: Add return type annotations to np.eye

### PR DESCRIPTION
**Pull request by Matt-Ord**

### CODE REVIEW

**Comment by Matt-Ord** [comment]
#18473

File: numpy/__init__.pyi [hunk 1]
```diff
@@ -408,7 +408,10 @@ dstack: Any
 ediff1d: Any
 expand_dims: Any
 extract: Any
-eye: Any
+eye: Callable[
```
Comment (BvB93) [inline numpy/__init__.pyi:411]: Merely annotating the function as `Callable` won't be enough to capture its signature unfortunately, as the former is unable to describe keyword arguments.

See `np.array` for a potential example: https://github.com/numpy/numpy/blob/eb4317d0a4c94b3f1454dca58c76caed8bbdc7e6/numpy/__init__.pyi#L2621-L2630

File: numpy/__init__.pyi [hunk 2]
```diff
@@ -575,9 +578,11 @@ _NdArraySubClass = TypeVar("_NdArraySubClass", bound=ndarray)
 _DTypeScalar_co = TypeVar("_DTypeScalar_co", covariant=True, bound=generic)
 _ByteOrder = Literal["S", "<", ">", "=", "|", "L", "B", "N", "I"]
 
+
```
Comment (BvB93) [inline numpy/__init__.pyi:581]: Can you remove all these unrelated edits?

**Review by BvB93 — CHANGES_REQUESTED** [review]

**Comment by Matt-Ord** [comment]
I have made the correct changes but accidentaly closed this pr- the changes can be seen at #18475
